client/packages.go

```diff
@@ -30,17 +30,26 @@ import (
 	"github.com/snapcore/snapd/snap"
 )
 
+// StoreAccount holds information about a store account.
+type StoreAccount struct {
+	ID          string `json:"id"`
+	Username    string `json:"username"`
+	DisplayName string `json:"display-name"`
+}
+
 // Snap holds the data for a snap as obtained from snapd.
 type Snap struct {
-	ID               string        `json:"id"`
-	Title            string        `json:"title,omitempty"`
-	Summary          string        `json:"summary"`
-	Description      string        `json:"description"`
-	DownloadSize     int64         `json:"download-size,omitempty"`
-	Icon             string        `json:"icon,omitempty"`
-	InstalledSize    int64         `json:"installed-size,omitempty"`
-	InstallDate      time.Time     `json:"install-date,omitempty"`
-	Name             string        `json:"name"`
+	ID            string        `json:"id"`
+	Title         string        `json:"title,omitempty"`
+	Summary       string        `json:"summary"`
+	Description   string        `json:"description"`
+	DownloadSize  int64         `json:"download-size,omitempty"`
+	Icon          string        `json:"icon,omitempty"`
+	InstalledSize int64         `json:"installed-size,omitempty"`
+	InstallDate   time.Time     `json:"install-date,omitempty"`
+	Name          string        `json:"name"`
+	Publisher     *StoreAccount `json:"publisher,omitempty"`
```

Comment:
Do we need to be concerned about `nil` here? I.e. I see in the PR that we have client code that uses snap.Publisher.Username - which might crash if Publisher is unset?

Reply:
it's never nil if the snap is known to come from the store. I think this makes for a saner JSON format, otoh slightly more delicate for the client